File: gadlift-backend/models/user.model.js
const mongoose = require("mongoose")
const bcrypt = require("bcrypt")
const userSchema = new mongoose.Schema({
    email: {
        type: String,
        required: true,
        unique: true,
        minlength: 6,
        trim: true
    },
    password: {
        type: String,
        required:true,
        minlength: 8
    },
    firstName: {
        type: String,
        required: true,
        minlength: 6
    },
    lastName: {
        type: String,
        required: true,
        minlength: 6
    },
    userType:{
        type:String,
        default:"user"
    }
})
userSchema.pre("save", async function (next) {
    try {
        const salt = await bcrypt.genSalt(10)
        this.password = await bcrypt.hash(this.password, salt)
        next()
    } catch (error) {
        console.log(error);
        next(error);
    }
},
    {
        timestamps: true,
    }
)
const User = mongoose.model("User", userSchema)
module.exports = User

File: mobile_app/lib/helpers/models/user.model.dart
class User {
  String id;
  String email;
  String first_name;
  String last_name;
  String profile_img;
  String mobile_number;

  User(
      {required this.id,
      required this.email,
      required this.first_name,
      required this.last_name,
      required this.profile_img,
      required this.mobile_number});

  factory User.fromJson(Map json) {
    return User(
        id: json['_id'],
        email: json['email'],
        first_name: json['first_name'],
        last_name: json['last_name'],
        mobile_number: json['mobile_number'],
        profile_img: json['profile_img']);
  }
}
